name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/actions/InsertAction.java

```diff
@@ -87,6 +98,38 @@ public Status process(BuySellEntry entry, Account account, Portfolio portfolio)
         // security via the dialog)
         process(entry.getPortfolioTransaction().getSecurity());
 
+        // when importing transactions that have already been generated by an
+        // investment plan, find the existing item and update it
+        if (investmentPlanItem)
+        {
+            DetectDuplicatesAction action = new DetectDuplicatesAction(client);
+            Transaction existingTransaction = null;
+            PortfolioTransaction t = entry.getPortfolioTransaction();
+
+            // search for a match in existing investment plan transactions
+            List<InvestmentPlan> plans = client.getPlans();
+            Iterator<InvestmentPlan> i = plans.stream().filter(p -> p.getSecurity().equals(t.getSecurity())).iterator();
+            while (i.hasNext())
+            {
+                List<Transaction> transactions = i.next().getTransactions();
+                existingTransaction = action.findInvestmentPlanTransaction(t, transactions);
```

Comment:
Shouldn't the *first* transaction break the loop?
Let's say I have two investment plans on the same security. And the duplicate is found in the first, then the second would overwrite it with null again.

Reply:
Good point. New version breaks and returns after `existingTransaction != null`.